### PR TITLE
Routing: Use location from hooks

### LINKED ISSUE #94066: Replace props.location with useLocation
Replace props.location with the useLocation hook.

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Replace the `location` prop with the `useLocation` hook.

**Why do we need this feature?**

Part of the react router migration.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/94066

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/AppWrapper.tsx [hunk 1]
```diff
@@ -131,3 +120,18 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
     );
   }
 }
+
+type Props = RouteComponentProps & { route: RouteDescriptor };
+
+function RouteComponent(props: Props) {
```
Comment (jackw) [inline public/app/AppWrapper.tsx:126]: Is this component necessary? Could `GrafanaRoute` use the location hook instead?

**Review by joshhunt — APPROVED** [review]

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**